### ISSUE TITLE
Second Plugin

### ISSUE THREAD

**Issue by TheKnightGamer** [ImagicalDevs/CosmeticMenu#1]
This well be the second plugin to start with.

**Comment by ImagicalGamer** [comment]
Just looking at the README makes me excited xP.

**Comment by TheKnightGamer** [comment]
OK guys let start with this first let make the thing when you have a compass (not a chest) you tap on a block it open a chest menu.

**Comment by TheKnightGamer** [comment]
@LittleBigMC  We have to do what the readme says

**Comment by LittleBigMC** [comment]
@TheKnightGamer im just adding the code for the gadgets. Bessides idk how to do addwindows. It dont work for me. And also i might have submitted this to the im forums but i put it as WIP so anyone knows we havnt done yet

**Comment by TheKnightGamer** [comment]
We are making a new account so we will upload on that .

**Comment by LittleBigMC** [comment]
Oh ok

**Comment by LittleBigMC** [comment]
Also @TheKnightGamer can u add code for chest open inventories after. I cant do that. I dont know how. And also what other gadgets should there be???

**Comment by TheKnightGamer** [comment]
Don't know what other gadget and I am trying to figure out the open chest stuff.

**Comment by LittleBigMC** [comment]
I know the code it is
$chest = $this->getServer()->getLevelByName("ChestWorld")->getTile(new
Vector3(1, 1, 1));
$player->addWindow($chest->getInventory)
But i dont know where to make it so that it will create a chest in that
position

**Comment by TheKnightGamer** [comment]
No we need a compass that when tap a block it open a chest menu and it dose not place a chest .

**Comment by LittleBigMC** [comment]
Ik but we need a chets somewhere in the world to get its inventory and open.
Add this:
If($item->getId() == 345){
$chest = $this->getServer()->getLevelByName("ChestWorld")->getTile(new
Vector3(1, 1, 1)); this will be where our chest it
$player->addWindow($chest->getInventory())

**Comment by TheKnightGamer** [comment]
OK that will work.

**Comment by LittleBigMC** [comment]
K but im only asking u to do cause it didnt work for me and maybe i did it wrong

**Comment by LittleBigMC** [comment]
Also we need to set the inventory of the chest and when thr player removes the object inside( the apple for the gadgets) it opens a new chest with the gadgets inside

**Comment by TheKnightGamer** [comment]
Yes we need that but I don't know much of open chest using items but there is one plugin that open a chest when touch a npc

**Comment by LittleBigMC** [comment]
Send me link i will read its code and i will do this its fine. I just need
to find that plugin. Do u have a link???

**Comment by TheKnightGamer** [comment]
No but I know the plugin name GoldSTD

**Comment by LittleBigMC** [comment]
@TheKnightGamer i have a new idea. Cause chest is too hard, we can make it in their inventories. When they hold down a compass, we give them an apple(for gadgets) and a blaze rod(for particles soon). When they hold down with the item, we remove them from inv and add new items! This will be much easier and a better way of doing it. What do u think???

**Comment by TheKnightGamer** [comment]
This is much better. Let start from start we need a way for a player when tap a block with a compass it gives you two more items let go find codes.

**Comment by LittleBigMC** [comment]
@TheKnightGamer i already know how to do it just asking if u liked or didnt like it. Also no need to start new. Ill work right now

**Comment by TheKnightGamer** [comment]
But we need to organized things you know that no messy things.

**Comment by LittleBigMC** [comment]
Ok wait first i write code and u tidy it up plz i dont want to start again.

**Comment by TheKnightGamer** [comment]
OK I will tidy things up but you can take the codes tidy it and then put it.

**Comment by LittleBigMC** [comment]
@TheKnightGamer already coded the compass thingy ;D it was quite easy tbh.

**Comment by TheKnightGamer** [comment]
Yeah it look so

**Comment by LittleBigMC** [comment]
Thanks

**Comment by TheKnightGamer** [comment]
It already looks tidy now let test it but where is your server php7

**Comment by LittleBigMC** [comment]
@TheKnightGamer i wanna remove this repo from imdevs and post on my profile as no one has been really helping. This plugin so far is only by me :/

**Comment by TheKnightGamer** [comment]
Yeah because you are speedy gandzales I can contribut but whatever I want to put you already there to do it and plus we are working on SG when that finish we come here.

**Comment by TheKnightGamer** [comment]
@Taki21 want to help out on this.

**Comment by LittleBigMC** [comment]
Ok lol umm sorry and also cancel particles only do gadgets or it becomes
laggy. Maybe something else like pets or something else.

**Comment by Taki21** [comment]
I will helps oon

**Comment by LittleBigMC** [comment]
Ok ty.

**Comment by TheKnightGamer** [comment]
Pets are more laggy and let cancel particles that future update.

**Comment by LittleBigMC** [comment]
Ok and yep pets are really laggy. It was just an idea. Any more???

**Comment by TheKnightGamer** [comment]
Not really but in the future let stick with gadgets.

**Comment by LittleBigMC** [comment]
I think that it wont work for php7 @TheKnightGamer cause ot uses Float not FloatTag. I will try to remake for php7

**Comment by TheKnightGamer** [comment]
OK pls do.

**Comment by LittleBigMC** [comment]
Ok i will try

**Comment by LittleBigMC** [comment]
Im not too good with converting so i will need someone else to do it.

**Comment by TheKnightGamer** [comment]
I will help to

**Comment by TheKnightGamer** [comment]
OK just change the nbt it was easy and I think its php7 proof

**Comment by TheKnightGamer** [comment]
The plugin look messy so I will try to fix it up a bit but not right now later.

**Comment by TheKnightGamer** [comment]
Oh yeah stay with two gadgets for now let's work on SG

**Comment by LittleBigMC** [comment]
@TheKnightGamer r u finished yet? If so, lets get started some time tommorow.

**Comment by TheKnightGamer** [comment]
No but just adding section then tomorrow is the features of those sections

**Comment by LittleBigMC** [comment]
Also check the sgu chat thingy. I said a heartful thing back there.

**Comment by LittleBigMC** [comment]
XD

**Comment by LittleBigMC** [comment]
@TheKnightGamer i made it now.

**Comment by LittleBigMC** [comment]
@TheKnightGamer u just deleted the egg launcher....

**Comment by TheKnightGamer** [comment]
Oh don't worry I will be adding things soon I have it all save.
